### PR TITLE
Python: Add support for Python 3.12

### PR DESCRIPTION
**Pull request by steinsgateted**
solve #8499

### CODE REVIEW

**Comment by Fokko** [comment]
I'm confused as to why it doesn't kick off the regular Python test suite 🤔

File: .github/workflows/python-ci.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@ jobs:
     runs-on: ubuntu-22.04
     strategy:
       matrix:
-        python: ['3.8', '3.9', '3.10', '3.11']
+        python: ['3.8', '3.9', '3.10', '3.11', '3.12'']
```
Comment (Fokko) [inline .github/workflows/python-ci.yml:42]: This might be the reason:
```suggestion
        python: ['3.8', '3.9', '3.10', '3.11', '3.12']
```

Comment (steinsgateted) [reply .github/workflows/python-ci.yml]: Thanks. Update.

**Comment by jayceslesar** [comment]
Does it make sense to modify the action to install from deadsnakes? Or is it expected that this fails until 3.12 is released

**Comment by Fokko** [comment]
@jayceslesar I prefer to have official actions, but we can test with the deadsnakes one.

I'm closing this PR, @steinsgateted if you're interested in working on this, please reopen the PR in the Python repository: https://github.com/apache/iceberg-python

**Comment by steinsgateted** [comment]
@Fokko Thank you, reopen the PR: [Python: Add support for Python 3.12](https://github.com/apache/iceberg-python/pull/35)

**Pull request closed without merging**